### PR TITLE
Log operator initial sync timings

### PR DESCRIPTION
**Pull request by cgwalters**

### CODE REVIEW

**Comment by cgwalters** [comment]
Timings from a CI run:
```
I0118 22:40:11.214606       1 sync.go:47] [init mode] synced pools in 35.793377ms
I0118 22:40:18.681254       1 sync.go:47] [init mode] synced mcs in 7.453014364s
I0118 22:40:32.455741       1 sync.go:47] [init mode] synced mcd in 13.704618657s
I0118 22:40:46.895299       1 sync.go:47] [init mode] synced mcc in 14.356922167s
I0118 22:40:54.927609       1 sync.go:47] [init mode] synced required-pools in 8.000336266s
I0118 22:40:54.947661       1 sync.go:58] Initialization complete
```

File: pkg/operator/sync.go [hunk 1]
```diff
@@ -19,26 +19,33 @@ import (
...
 	}
 
 	var errs []error
-	for _, f := range syncFuncs {
-		errs = append(errs, f(rconfig))
+	for _, sf := range syncFuncs {
+		startTime := time.Now()
+		errs = append(errs, sf.fn(rconfig))
+		if optr.inClusterBringup {
+			glog.Infof("[init mode] synced %s in %v", sf.name, time.Since(startTime))
```
Comment (jlebon) [inline pkg/operator/sync.go]: Oh this is neat!

**Comment by cgwalters** [comment]
I'm lifting WIP on this since I think it's an improvement.

File: pkg/operator/sync.go (hunk 1 above)
Comment (abhinavdahiya) [reply pkg/operator/sync.go]: why report only on cluster bringup ?

Comment (cgwalters) [reply pkg/operator/sync.go]: Currently there are some `.V(4)` level logs in `sync()`.  It wasn't clear to me at what cadence we'd end up logging.  Glancing at the operator...it watches all daemonsets in our namespace, so we'll do a sync every time e.g. a node updates and the MCD state changes right?

I guess we can do more sync logging if we more consistently add some rate-limiting/delays like https://github.com/openshift/machine-config-operator/pull/337 ?

**Comment by cgwalters** [comment]
OK I rebased :surfing_man: and dropped the reordering.  This one now just adds "initial sync time" logging.

**Comment by jlebon** [comment]
This looks good to me. Will let @abhinavdahiya have another look.
/approve

**Comment by abhinavdahiya** [comment]
> This looks good to me. Will let @abhinavdahiya have another look.
> /approve

will defer to the team, I have no strong opinions.

**Comment by cgwalters** [comment]
Looking at the logs in this PR, we kept printing "Initialization complete".  Fixed and rebased :surfing_man:

**Pull request merged**